File: apollo-portal/src/main/java/com/ctrip/framework/apollo/portal/service/GlobalSearchValueService.java
/*
 * Copyright 2024 Apollo Authors
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 */
package com.ctrip.framework.apollo.portal.service;

import com.ctrip.framework.apollo.common.dto.ItemInfoDTO;
import com.ctrip.framework.apollo.portal.api.AdminServiceAPI;
import com.ctrip.framework.apollo.portal.entity.vo.ItemInfo;
import com.ctrip.framework.apollo.portal.environment.Env;
import org.springframework.stereotype.Service;

import java.util.ArrayList;
import java.util.List;

@Service
public class GlobalSearchValueService {

    private final AdminServiceAPI.ItemAPI itemAPI;

    public GlobalSearchValueService(AdminServiceAPI.ItemAPI itemAPI) {
        this.itemAPI = itemAPI;
    }

    public List<ItemInfo> get_PerEnv_ItemInfo_BySearch(Env env, String key, String value) {
        List<ItemInfo> perEnvItemInfos = new ArrayList<>();
        List<ItemInfoDTO> perEnvItemInfoDTOs = itemAPI.getPerEnvItemInfoBySearch(env, key, value);
        perEnvItemInfoDTOs.forEach(itemInfoDTO -> {
            ItemInfo itemInfo = new ItemInfo(itemInfoDTO.getAppId(),env.getName(),itemInfoDTO.getClusterName(),itemInfoDTO.getNamespaceName(),itemInfoDTO.getStatus(),itemInfoDTO.getKey(),itemInfoDTO.getValue());
            perEnvItemInfos.add(itemInfo);
        });
        return perEnvItemInfos;
    }

}

File: apollo-portal/src/main/resources/static/scripts/services/GlobalSearchValueService.js
/*
 * Copyright 2024 Apollo Authors
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 */
appService.service('GlobalSearchValueService', ['$resource', '$q', 'AppUtil', function ($resource, $q, AppUtil) {
    let global_search_resource = $resource('', {}, {
        find_different_env_properties_item_info: {
            method: 'GET',
            isArray: true,
            url: AppUtil.prefixPath() + '/global-search/properties/by-env-or-key-or-value?env=:env&key=:key&value=:value',
        },
        find_different_env_exclude_properties_item_info: {
            method: 'GET',
            isArray: true,
            url: AppUtil.prefixPath() + '/global-search/exclude-properties/by-env-or-value?env=:env&value=:value',
        },
    });
    return {
        findDifferentEnvPropertiesItemInfo:function (env,key,value){
            let d = $q.defer();
            global_search_resource.find_different_env_properties_item_info({env: env,key: key,value: value},function (result) {
                d.resolve(result);
            }, function (result) {
                d.reject(result);
            });
            return d.promise;
        },
        findDifferentEnvExcludePropertiesItemInfo:function (env,value){
            let d = $q.defer();
            global_search_resource.find_different_env_exclude_properties_item_info({env: env,value: value}, function (result) {
                d.resolve(result);
            }, function (result) {
                d.reject(result);
            });
            return d.promise;
        }
    }
}]);
